gems/sorbet-runtime/lib/types/compatibility_patches.rb

```diff
@@ -36,3 +36,27 @@ def observe!(method_name)
     end
   end
 end
+
+module T
+  module CompatibilityPatches
+    module MethodExtensions
+      def arity
+        arity = super # rubocop:disable PrisonGuard/NoArity
+        return arity if arity != -1 || self.is_a?(Proc)
+        sig = T::Private::Methods.signature_for_method(self)
+        sig ? sig.method.arity : arity # rubocop:disable PrisonGuard/NoArity
+      end
+
+      def source_location
+        sig = T::Private::Methods.signature_for_method(self)
+        sig ? sig.method.source_location : super
+      end
+
+      def parameters
+        sig = T::Private::Methods.signature_for_method(self)
+        sig ? sig.method.parameters : super
+      end
+    end
+    ::Method.prepend(MethodExtensions)
```

Comment:
what do you think about omitting the prepend from sorbet-runtime, only defining the module that could be prepended, and having people call prepend instead of calling some T::Configuration thing that calls a prepend?